clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -381,8 +394,8 @@ public ConfigKey(String name, Type type, Object defaultValue, Validator validato
             this.defaultValue = defaultValue;
             this.validator = validator;
             this.importance = importance;
-            if (this.validator != null)
-                this.validator.ensureValid(name, defaultValue);
+            if (hasDefault() && validator != null)
```

Comment:
Well, what about the validation such that the default value must be an empty string?